### PR TITLE
Move code from RemoteSession to SessionRepository

### PR DESCRIPTION
**Pull request by hmusum**
Simplify and move in the direction of unifying remote and local sessions
Move tests and avoid a lot of low-level setup code

### CODE REVIEW

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/RemoteSession.java [hunk 1]
```diff
@@ -26,95 +15,46 @@
...
 
-    public synchronized ApplicationSet ensureApplicationLoaded() {
-        return applicationSet == null ? applicationSet = loadApplication() : applicationSet;
+    Optional<ApplicationSet> applicationSet() {
+        return applicationSet;
     }
 
-    public synchronized void deactivate() {
-        applicationSet = null;
+    public synchronized RemoteSession deactivate() {
```
Comment (jonmv) [inline configserver/src/main/java/com/yahoo/vespa/config/server/session/RemoteSession.java:52]: Would like a name change here, to `deactivated()`

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java [hunk 2]
```diff
@@ -275,13 +280,25 @@ public int deleteExpiredRemoteSessions(Clock clock, Duration expiryTime) {
...
+                deleteSession(session);
                 deleted++;
             }
         }
         return deleted;
     }
 
+
+    public void deactivate(RemoteSession remoteSession) {
+        remoteSessionCache.addSession(remoteSession.deactivate());
```
Comment (jonmv) [inline configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java:292]: `addSession` should perhaps also be named `putSession`?

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java [hunk 3]
```diff
@@ -358,27 +375,19 @@ void activate(RemoteSession session) {
...
     }
 
-    public void deactivate(RemoteSession remoteSession) {
-        remoteSession.deactivate();
-    }
-
-    public void delete(RemoteSession remoteSession, Optional<LocalSession> localSession) {
+    public void deleteSession(RemoteSession remoteSession, Optional<LocalSession> localSession) {
         localSession.ifPresent(this::deleteLocalSession);
         remoteSession.deactivate();
```
Comment (jonmv) [inline configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java:388]: Shouldn't this also be `deactivate(remoteSession)`? This does nothing now.

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java [hunk 4]
```diff
@@ -394,13 +403,84 @@ private void loadSessionIfActive(RemoteSession session) {
...
+        RemoteSession newSession = new RemoteSession(session.getTenantName(),
+                                    session.getSessionId(),
+                                    session.getSessionZooKeeperClient(),
+                                    Optional.of(newApplicationSet));
+        remoteSessionCache.addSession(newSession);
+        return newApplicationSet;
+    }
+
+    void confirmUpload(RemoteSession session) {
+        Curator.CompletionWaiter waiter = session.getSessionZooKeeperClient().getUploadWaiter();
```
Comment (jonmv) [inline configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java:436]: Why `session.getSessionZooKeeperClient()` here, and `createSessionZooKeeperClient(session.getSessionId())` elsewhere?

**Review by jonmv — CHANGES_REQUESTED** [review]
Yes, this will be much better :)

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java (hunk 4 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java:436]: Yes, both are used, I want to clean up that, will do in a later PR

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java (hunk 2 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java:292]: Agreed, will fix

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/RemoteSession.java (hunk 1 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/session/RemoteSession.java:52]: 👍

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java (hunk 3 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java:388]: Yes, thanks

**Review by jonmv — APPROVED** [review]

**Pull request merged**